frontend/src/services/sms.service.ts

```diff
@@ -30,29 +30,34 @@ export async function saveTemplate(
   }
 }
 
-export async function validateNewCredentials({
+export async function validateAndStoreNewCredentials({
   campaignId,
+  label,
   accountSid,
   apiKey,
   apiSecret,
   messagingServiceSid,
   recipient,
 }: {
   campaignId: number
+  label: string
   recipient: string
   accountSid: string
   apiKey: string
   apiSecret: string
   messagingServiceSid: string
 }): Promise<void> {
   try {
-    await axios.post(`/campaign/${campaignId}/sms/new-credentials`, {
+    await storeCredentials({
```

Comment:
@miazima can you explain again why we've chose to split up storing and validation of the credentials into separate API calls? I think you've explained this before but it has slipped my mind (sorry!). 

My concern is that we are storing credentials into SecretsManager without first validating them. In the current flow, if I pass in an invalid credential, it will be stored into SecretsManager even though an error is thrown eventually. Over time we could have a bunch of invalid creds in SecretsManager that will be hard to clean up.

Reply:
I see your concern, my first approach was to modify the `/new-credentials` endpoint to require label as one of the parameters - this would have been a breaking change for API users and making it an optional param would mean that `checkUserCredentialLabel` and `storeUserCredential` middleware would have to handle undefined `label` values which defeats the purpose of those middleware.

Hence we chose to split up the storing and validation calls. I understand setting `validate` at the settings route may not be a good idea. If we move this optional validate params to `/campaign/{campaignId}/sms/credentials` I believe there would not be any unintended side effects. `validate === false` feels hackish, but I couldn't think of an alternative solution to this.

Reply:
@miazima I've tried to briefly refactor to see if we can conditionally apply the middlewares for saving user credentials. It's on the `feature/save-credentials-alt` branch. You can check out the diffs at [here](https://github.com/opengovsg/postmangovsg/compare/feature/save-credentials...feature/save-credentials-alt). Not sure if it's cleaner though.

Reply:
I think it is a more straightforward way of implementing this conditional instead of modifying the middlewares, esp since `validateAndStoreCredentials` is shared by multiple endpoints, which would be confusing. Including comments explaining the need for the new middleware would make this clearer too. Feel free to rebase your changes on this branch.
In `storeUserCredential`, we could return `next()` if `campaignId` params exists, else return `return res.json({ message: 'OK' })`